src/amqproxy.cr

```diff
@@ -71,9 +74,14 @@ class AMQProxy::CLI
     port = u.port || default_port
     tls = u.scheme == "amqps"
 
-    server = AMQProxy::Server.new(u.host || "", port, tls, @log_level, @idle_connection_timeout)
+    server = AMQProxy::Server.new(u.host || "", port, tls, @log_level, @idle_connection_timeout, @graceful_shutdown)
 
+    @got_sigterm = false
     shutdown = ->(_s : Signal) do
+      if @got_sigterm
+        return
+      end
+      @got_sigterm = true
```

Comment:
I'm pretty sure we want a second SIGINT (^C) to shutdown amqproxy instantly (most programs work like that)

Reply:
Implemented this